packages/vm/src/cache.rs

```diff
@@ -1409,6 +1430,25 @@ mod tests {
         );
     }
 
+    #[test]
+    fn heavy_metrics_works() {
+        let cache = unsafe { Cache::new(make_testing_options()).unwrap() };
+        let checksum = cache.save_wasm(CONTRACT).unwrap();
+
+        cache.pin(&checksum).unwrap();
+
+        let heavy_metrics = cache.heavy_metrics();
+        assert_eq!(heavy_metrics.hits_per_pinned_contract, [(checksum, 0)]);
+
+        let backend = mock_backend(&[]);
+        let _ = cache
+            .get_instance(&checksum, backend, TESTING_OPTIONS)
+            .unwrap();
+
+        let heavy_metrics = cache.heavy_metrics();
+        assert_eq!(heavy_metrics.hits_per_pinned_contract, [(checksum, 1)]);
```

Comment:
Could you add a second contract here too to have a case with > 1 element?

Reply:
Added a case with two contracts. The checks of the hit count had to be modified a bit since the `PinnedMemoryCache` is using a regular `HashMap` internally, making the order of the keys non-deterministic.